### ISSUE TITLE
Total pages and redirect to a specific page

### ISSUE THREAD

**Issue by gonzalez-aldo** [futurepress/epub.js#1252]
I can´t find a way to get the total pages of a whole epub file, let alone redirect to a specific page. I know that this depends on the epub's chapters and his rendition (with his size letter and others settings) but there should be a way to calculate it.

**Comment by OriIdan** [comment]
To get the total number of pages use: this.book.locations.length()
This is not exactly pages, this is the number of locations calculated at
the beginning (by default every 600 characters).
To jump to specific page
use this.book.locations.cfiFromLocation(pageNumber) This will return the
CFI of the page. To actually jump use: this.display(cfi);

I hope this helps.



-- 
Ori Idan CEO Helicon Books
http://www.heliconbooks.com

**Comment by RatulSikder97** [comment]
Another way to get total Page
1. Hiddenly render every chapter.
2. Sum up the value of ` rendition.location.start.displayed.total`
3. Total summation is the total page by rendition format.
4. Finally destroy hidden Rendition and elements.

**Comment by Erdogan34** [comment]
@OriIdan this.book.locations.cfiFromLocation(pageNumber) always return -1, what can be the reason?

**Comment by OriIdan** [comment]
Make sure you have locations.
Locations should be calculated at first opening.

**Comment by Erdogan34** [comment]
@OriIdan thanks for your answer. How can I calculate it at first, could you share a sample, please?

**Comment by OriIdan** [comment]
Use: this.book.locations.generate(600);
You should call locations.generate when you open the book, you can also
save the result. The result is an array of objects.
In this example it will create a "page" every 600 characters. It does not
affect  the actual paging according to font size and screen size.

**Comment by Erdogan34** [comment]
@OriIdan yes its returning now! but unfortunately not accurate with loc.start.displayed.page value. 

If I enter the page number as 10, it navigated to page 6.

![image](https://user-images.githubusercontent.com/7986233/156925182-cda2624c-d932-45b7-aaf5-fea05476d39c.png)

**Comment by gonzalez-aldo** [comment]
> Another way to get total Page
> 
> 1. Hiddenly render every chapter.
> 2. Sum up the value of ` rendition.location.start.displayed.total` of each chapter
> 3. Total summation is the total page by rendition format.
> 4. Finally destroy hidden Rendition and elements.

This work for me. But it's to slowly.. Any way to get every chapter faster?
